packages/test/stochastic-test-utils/src/generators.ts

```diff
@@ -148,27 +149,39 @@ export function chainIterables<T, TState>(
 	};
 }
 
+/**
+ * Controls exit behavior for {@link interleave}.
+ */
+export enum ExitBehavior {
+	OnBothExhausted,
+	OnEitherExhausted,
+}
+
 /**
  * Interleaves outputs from `generator1` and `generator2`.
  * By default outputs are taken one at a time, but can be controlled with `numOps1` and `numOps2`.
  * This is useful in stochastic tests for producing a certain operation (e.g. "validate" or "synchronize") at a
  * defined interval.
  *
- * Exhausts both input generators before terminating.
+ * Exhausts both input generators before terminating by default. If {@link ExitBehavior.OnEitherExhausted} is
+ * provided, instead exits as soon as the next element it would produce is `done`.
  *
  * @example
  * ```typescript
  * // Assume gen1 produces 1, 2, 3, ... and gen2 produces "a", "b", "c", ...
  * interleave(gen1, gen2) // 1, a, 2, b, 3, c, ...
  * interleave(gen1, gen2, 2) // 1, 2, a, 3, 4, b, 5, 6, c, ...
- * interleave(gen1, gen2, 2, 3) // 1, 2, a, b, c, 3, 4, d, e, f, ...
+ * interleave(take(2, gen1), gen2, 1, 1, ExitBehavior.OnEitherExhausted) // 1, a, 2, b
```

Comment:
Not a big deal since the description above is accurate, but the example in the test is clearer to me that it's not this:
interleave(gen1, take(2, gen2), 1, 1, ExitBehavior.OnEitherExhausted) // 1, a, 2, b
but this:
interleave(gen1, take(2, gen2), 1, 1, ExitBehavior.OnEitherExhausted) // 1, a, 2, b, 3